Add Instant micro methods to Typescript definitions.

mattbishop opened the pull request:
These methods were added to js-joda recently, and the typescript definition should match.

pithu commented:
Hi @mattbishop thanks for contributing, could you please add a basic test at packages/core/test/typescript_definitions

mattbishop commented:
Thanks for the request, I added typings tests in the 'Instant' test suite. I also added in flow method declarations, though I could not find any tests for those.
